app/models/airing.js

```diff
@@ -13,20 +14,19 @@ export default DS.Model.extend({
   playlistDaily: belongsTo({ async: false }),
   // episode: belongsTo('story', { async: false }),
   tracks: hasMany('tracks', { async: false, inverse: 'airing' }),
-  isCurrent: true,
-  isLive: false,
+  currentTime: new Date(),
 
   setTime(currentTime) {
-    let isLive      = (currentTime >= new Date(this.startTime) && currentTime <= new Date(this.endTime));
-    let isCurrent   = isLive || (currentTime > new Date(this.endTime));
-
-    if (isLive != this.isLive) {
-      this.set('isLive', isLive);
-      this.notifyPropertyChange('isLive');
-    }
-    if (isCurrent != this.isCurrent) {
-      this.set('isCurrent', isCurrent);
-      this.notifyPropertyChange('isPast');
-    }
-  }
+    this.set('currentTime', currentTime);
+  },
+  isLive: computed('currentTime', 'startTime', 'endTime', function() {
+    return this.currentTime >= new Date(this.startTime) && this.currentTime <= new Date(this.endTime);
+  }),
+  isCurrent: computed('isLive', 'currentTime', 'startTime', 'endTime', function() {
+    let isPast = this.currentTime > new Date(this.endTime);
+    let isFuture = this.currentTime < new Date(this.startTime);
+    return this.isLive || isPast || isFuture;
+  }),
```

Comment:
I don't see `isCurrent` being called anywhere, anymore. Is it still needed now?

Reply:
`this.isLive || isPast || isFuture` == always true, right?

Reply:
@bkammin -- `isCurrent` is used in the play-schedule's template.hbs  -- or did I misuse this?

Reply:
yeah I think this will always be true. Is current means that it's now or in the future but the question is what is isCurrent meant to do?